packages/typespec-python/src/emitter.ts

```diff
@@ -539,16 +539,32 @@ function isAzureCoreModel(t: Type): boolean {
     );
 }
 
+function hasDefaultStatusCode(response: HttpOperationResponse): boolean {
+    return response.statusCode === "*";
+}
+
 function emitResponse(
     context: SdkContext,
     response: HttpOperationResponse,
     innerResponse: HttpOperationResponseContent,
+    operation: Operation,
 ): Record<string, any> {
     let type = undefined;
+    let resultProperty = undefined;
     if (innerResponse.body?.type) {
         let modelType = undefined;
         if (innerResponse.body.type.kind === "Model") {
             modelType = getEffectiveSchemaType(context, innerResponse.body.type);
+            const lroMeta = getLroMetadata(context.program, operation);
+            // only when getLroMetadata can provide effective model with valid name
+            if (!hasDefaultStatusCode(response) && lroMeta && lroMeta.logicalResult.name) {
```

Comment:
replied https://github.com/Azure/autorest.python/pull/1821/files#r1174000085
@iscai-msft Before https://github.com/Azure/typespec-azure/issues/2862 fixed, we can use current logic to avoid issue in rpc testcase. And there is other merits: getLroMetadata may return invalid response type beyond rpc testcase and we can avoid that potential problem by checking if it has valid name.

Reply:
@iscai-msft As discussed in scrum meeting, I hide the rpc testcase and will open it again after getLroMetadata issue fixed. Please merge this PR.